integration/spark-extension-columnlevel/src/main/java/io/openlineage/spark/extension/column/v1/DatasetIdentifier.java

```diff
@@ -0,0 +1,107 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+package io.openlineage.spark.extension.column.v1;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+
+public class DatasetIdentifier {
+  private final String name;
+  private final String namespace;
+  private final List<Symlink> symlinks;
+
+  public enum SymlinkType {
+    TABLE
```

Comment:
Consider using `String` so that we won't need `v2` to support different symlink types.